demo/header-footer.html

```diff
@@ -0,0 +1,99 @@
+<html>
+
+<head>
+  <title>Fallback element</title>
+  <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1, user-scalable=yes">
+
+  <style>
+    body {
+      font-family: Roboto, Helvetica, sans-serif;
+      margin: 0;
+    }
+
+    virtual-list {
+      width: 100vw;
+      height: 100vh;
+    }
+
+    header {
+      padding: 10px;
+      box-sizing: border-box;
+      width: 100%;
+      color: #333;
+      background-color: #ddd;
+    }
+  </style>
+</head>
+
+<body>
+
+  <virtual-list></virtual-list>
+
+  <script type="module">
+    import '../virtual-list-element.js';
+    import './contacts/contact-element.js';
+
+    const virtualList = document.querySelector('virtual-list');
+
+    const nodePool = { group: [], contact: [] };
+    virtualList.newChild = (item) => {
+      let element;
+      if (item.group) {
+        element = nodePool.group.pop() || document.createElement('header');
+      } else {
+        element = nodePool.contact.pop() || document.createElement('contact-element');
+      }
+      return element;
+    };
+    virtualList.updateChild = (element, item) => {
+      if (item.group) {
+        element.textContent = `${item.title} (${item.count})`;
+      } else {
+        element.contact = item;
+      }
+    };
+    virtualList.recycleChild = (element, item) => {
+      if (item.group) {
+        nodePool.group.push(element);
+      } else {
+        nodePool.contact.push(element);
+      }
+    };
+
+    // Get contacts, sort and group by name.
+    fetch('contacts/contacts.json')
```

Comment:
Expand the comment here to explain what the input looks like and what the output will look like.